doc/source/user_guide/scale.rst

```diff
@@ -0,0 +1,362 @@
+.. _scale:
+
+*************************
+Scaling to large datasets
+*************************
+
+Pandas provide data structures for in-memory analytics. This makes using pandas
+to analyze larger than memory datasets somewhat tricky.
+
+This document provides a few recommendations for scaling to larger datasets.
+It's a complement to :ref:`enhancingperf`, which focuses on speeding up analysis
+for datasets that fit in memory.
+
+But first, it's worth considering *not using pandas*. Pandas isn't the right
+tool for all situations. If you're working with very large datasets and a tool
+like PostgreSQL fits your needs, then you should probably be using that.
+Assuming you want or need the expressivity and power of pandas, let's carry on.
+
+.. ipython:: python
+
+   import pandas as pd
+   import numpy as np
+   from pandas.util.testing import make_timeseries
```

Comment:
Just thinking through implications but is this something we really want to do? I feel like this is an unnecessary API exposure

Reply:
Open to suggestions here. I've tried to make the API exposure as small as possible. But I need a way to

1. Generate a semi-realistic dataset of arbitrary size.
2. Without distracting from the overall message.

So I think the options are what I have here, or a private method like `_make_timeseries()` and the docs just describes the raw data contained in the file on disk (but we hide the generation of that file).

Reply:
I think private method and hiding import would be preferable; maybe just a comment preceding first usage saying `# arbitrary large frame` or something to the effect. The user shouldn't care about the import machinery

Reply:
Just to be clear, if I make it private, I'm not going to show it being imported.

Reply:
Though I'll note in passing that a way to generate realistic, sample random datasets is nice :)